cli/dcoscli/node/main.py

```diff
@@ -65,12 +72,303 @@ def _cmds():
+    """
+    Get an requests HTTP response, read it and deserialize to json.
+
+    :param http_response: http response
+    :type http_response: requests.Response onject
+    :return: deserialized json
+    :rtype: dict
+    """
+    data = b''
+    try:
+        for chunk in http_response.iter_content(1024):
+            data += chunk
+        snapshot_response = util.load_jsons(data.decode('utf-8'))
+        return snapshot_response
+    except DCOSException:
+        raise
+
+
+def _snapshot_download(snapshot, location):
+    """
+    Download snapshot and put in the the current directory.
+
+    :param snapshot: snapshot file name.
+    :type snapshot: string
+    :return: exit code
+    :rtype: int
+    """
+
+    # make sure the requested snapshot exists
+    snapshot_size = 0
+    for available_snapshot in _get_snapshots_list():
+        if len(available_snapshot) != 2:
+            raise DCOSException('Incorrect snapshot response')
+
+        if available_snapshot[0] == snapshot:
+            snapshot_size = available_snapshot[1]
+
+    url = SNAPSHOT_BASE_URL + '/serve/' + snapshot
+    snapshot_location = os.path.join(os.getcwd(), snapshot)
+    if location:
+        if os.path.isdir(location):
+            snapshot_location = os.path.join(location, snapshot)
+        else:
+            snapshot_location = location
+
+    if snapshot_size > SNAPSHOT_WARN_SIZE:
+        if not confirm(
+            'Snapshot size is {}, do you want to download it?'.format(
+                sizeof_fmt(snapshot_size)), False):
+            return 0
+
+    r = _do_request(url, 'GET', stream=True, timeout=120)
+    try:
+        with open(snapshot_location, 'wb') as f:
+            for chunk in r.iter_content(1024):
+                f.write(chunk)
+    except Exception as e:
+        raise DCOSException(e)
+    emitter.publish('snapshot downloaded to ' + snapshot_location)
+    return 0
+
+
+def _snapshot_delete(snapshot):
+    '''
+    Delete a snapshot
+
+    :param snapshot: snapshot file name
+    :type: str
+    :return: exit code
+    :rtype: int
+    '''
+    _check_3dt_version()
+    response = _do_snapshot_request(
+        SNAPSHOT_BASE_URL + '/delete/' + snapshot, 'POST')
+    if 'status' not in response:
+        raise DCOSException('Incorrect response: {}'.format(response))
+    emitter.publish(response['status'])
+    return 0
+
+
+def _snapshot_create(nodes):
+    """
+    Create a snapshot.
+
+    :param nodes: a list of nodes to collect the logs from.
+    :type nodes: list
+    :returns: process return code
+    :rtype: int
+    """
+    _check_3dt_version()
+    response = _do_snapshot_request(SNAPSHOT_BASE_URL + '/create',
+                                    'POST',
+                                    json={'nodes': nodes})
+    if ('status' not in response or 'extra' not in response
+            or 'snapshot_name' not in response['extra']):
+        raise DCOSException('Incorrect response: {}'.format(response))
```

Comment:
Lets be a little more verbose for the user. Something like: "Request to create snapshot [request] returned an unexpected response [response]"